From c61e326ab8a3a7c9ca0d06e3ff33dad2e9d2e18e Mon Sep 17 00:00:00 2001
From: Mathias Fiedler <mathias_fiedler@mentor.com>
Date: Mon, 17 Aug 2015 13:18:31 +0200
Subject: [PATCH 24/26] waylandsink: don't drop buffers even when framecallback
 is still pending

Alternatively we could drop this buffer and send the release event.
---
 ext/wayland/gstwaylandsink.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index bb45c2a..9b2685e 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -980,11 +980,13 @@ gst_wayland_sink_render (GstBaseSink * bsink, GstBuffer * buffer)
     sink->video_info_changed = FALSE;
   }
 
+#ifndef USE_OMX_TARGET_VENUS
   /* drop buffers until we get a frame callback */
   if (g_atomic_int_get (&sink->redraw_pending) == TRUE) {
     GST_DEBUG_OBJECT(sink, "Dropping buffer %p while waiting on frame callback", buffer);
     goto done;
   }
+#endif
 
   if (G_UNLIKELY (sink->video_info_changed)) {
     gst_wl_window_set_video_info (sink->window, &sink->video_info);
-- 
1.9.1

